### PR TITLE
[Task] Create Functionality for Chatbot

### PR DESCRIPTION
**Pull request by Teraaay**
## Related Links

Task Link: https://www.notion.so/Create-functionality-for-chatbot-78513f47b9a747d0860ab7960340c6b1

## Description

- The chatbot should respond according to what was being sent by the user
- The messages should be saved and should still display after closing and opening the chatbot modal

## Notes

Here are the keywords with corresponding responses from the chatbot:
- options
- help
- question
- do for me
- adopt
- process
- paperwork
- document
- requirement

## Screenshots
chatbot

### CODE REVIEW

**Review by ume45876139 — APPROVED** [review]
# Working
## Chrome
<details>
<summary>PC video</summary>

https://user-images.githubusercontent.com/39412730/167585810-969d45f8-19c5-4e6c-98e9-5db2cf61168e.mov

</details>

<details>
<summary>Tablet</summary>

Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2022-05-10 at 17 42 09

</details>

<details>
<summary>iPhone video</summary>

https://user-images.githubusercontent.com/39412730/167586327-fd0753fc-0e81-415b-8e78-b700b1661d97.MP4

</details>

- Chatbot is working properly

[note]
- When user scroll up while open chat box, sometimes background is moving
    - It's very small bug so we can disregard for now

<details>
<summary>Small bug</summary>


https://user-images.githubusercontent.com/39412730/167586940-341e886e-4b0a-45ae-909a-91a4ab88bada.mov


</details>


# Testing data
test device: Chrome, Safari (Mac)
test time: 0.5h
comment: 0
iteration: 0
total test time: 0.5h

**Comment by roseaugusto** [comment]
@Teraaay I have two confirmation so far.

1. Is it possible to merge `src/pages/components/chatbot/ChatBox.js` and `src/pages/components/chatbot/index.js`? 
2. Please disable/prevent sending blank message. (is it doable?)

https://user-images.githubusercontent.com/70557829/167761989-a1f4bc36-fd2c-4469-837a-558ef6fa031e.mov

File: src/pages/components/chatbot/configs/chatbotConfig.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  if (!messages) {
+    localStorage.setItem('chat_messages', JSON.stringify(initialMessages));
+  }
+
+  return {
+    initialMessages,
+    customComponents: {
+      header: () => (
+        <div className="h-16 w-100 flex justify-between items-center bg-gradient-to-b from-sky-900 via-sky-700 to-mocha-500 rounded-t-lg px-3">
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/configs/chatbotConfig.js]:
```suggestion
        <div className='h-16 w-100 flex justify-between items-center bg-gradient-to-b from-sky-900 via-sky-700 to-mocha-500 rounded-t-lg px-3'>
```

use single quotes for classNames

File: src/pages/components/chatbot/configs/chatbotConfig.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          <img
+            className="rounded-full w-10"
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/configs/chatbotConfig.js]:
```suggestion
            className='rounded-full w-10'
```
use single quotes for classNames

File: src/pages/components/chatbot/configs/chatbotConfig.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            src="/images/RoundOrangeLogo.svg"
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/configs/chatbotConfig.js]: Please use single quotes

File: src/pages/components/chatbot/configs/chatbotConfig.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            alt="logo"
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/configs/chatbotConfig.js]: Please use single quotes

File: src/pages/components/chatbot/configs/chatbotConfig.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+          src="/images/ChatbotAvatar.svg"
+          alt="chatbot avatar"
+        />
+      ),
+
+      userAvatar: () => (
+        <img
+          className="border rounded-full w-10 h-10"
+          src="/images/UserAvatar.svg"
+          alt="user avatar"
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/configs/chatbotConfig.js]: Please use single quotes

File: src/pages/components/chatbot/index.js [hunk 6]
```diff
@@ -4,21 +4,25 @@ import ChatBox from './ChatBox';
...
-        className='z-50 cursor-pointer fixed bottom-0 right-0 mb-3 mr-6 w-[50px] md:mb-6 md:mr-[3.5rem] lg:w-[4.5rem] lg:my-4'
+        className="z-50 cursor-pointer fixed bottom-0 right-0 mb-4 mr-6 w-[50px] md:mb-7 md:mr-[2rem] lg:w-[4.5rem] lg:mb-7"
         onClick={() => setIsChatBoxOpen(!isChatBoxOpen)}
         src={
           isChatBoxOpen
             ? '/images/CloseFloatingButton.svg'
             : '/images/ChatFloatingButton.svg'
         }
-        alt='chat floating button'
+        alt="chat floating button"
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/index.js]: Please use single quotes

File: src/pages/components/chatbot/widgets/Options.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+const Options = ({ options }) => {
+  return (
+    <div className="pb-2">
+      <div className="flex flex-col mx-12">
+        {options.map((option) => {
+          return (
+            <div
+              className="mt-2 px-2 py-1 border-2 border-navy-200 rounded-lg cursor-pointer text-base text-center hover:bg-sky-100"
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/widgets/Options.js:10]: Please use single quotes

File: src/pages/components/chatbot/widgets/Process.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import { adoptionProcess } from '../../../../utilities/constants';
+
+const Process = () => {
+  return (
+    <div className="border-2 p-5 pb-0 mx-12 bg-sky-100 rounded-lg">
+      <ul>
+        {adoptionProcess.map((process) => {
+          return (
+            <Fragment key={process.id}>
+              <h1 className="font-bold pb-2">{process.title}</h1>
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/widgets/Process.js:11]: Please use single quotes

File: src/pages/components/chatbot/widgets/Process.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+              <li className="pb-5">{process.text}</li>
+            </Fragment>
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/widgets/Process.js:13]: Please use single quotes

File: src/pages/components/chatbot/widgets/Requirements.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { requirements } from '../../../../utilities/constants';
+
+const Requirements = () => {
+  return (
+    <div className="border-2 p-4 pl-9 mx-12 bg-sky-100 rounded-lg">
+      <ul className=" list-disc">
+        {requirements.map((req) => {
+          return (
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/widgets/Requirements.js:9]: Please use single quotes

File: src/pages/components/chatbot/widgets/Requirements.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+            <li className="pb-3" key={req.id}>
```
Comment (CaneteJoash) [inline src/pages/components/chatbot/widgets/Requirements.js:10]: Please use single quotes

**Review by roseaugusto — APPROVED** [review]
Approved!

**Pull request merged**